### DISCUSSION TITLE
How to plot PR curves for all the IoU thresholds?

### DISCUSSION THREAD

**Discussion by dkobayas-cyber** [ultralytics/ultralytics#1519] [Q&A]
Hello,
Is there any quick way to plot the precision-recall curve for ten IoU thresholds ranging from 0.5 to 0.95? 
In [/ultralytics/ultralytics/yolo/utils/metric.py](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/yolo/utils/metrics.py), the function `ap_per_class` seems to be computing some relevant metrics. I'm wondering if I can somehow modify the function to plot the ten different PR curves for each class.

**Comment by glenn-jocher** [comment]
@dkobayas-cyber you could customize the code for this, but I believe by default the curve plots are created at 0.50 IoU
